### PR TITLE
05.03.01

### PR DESCRIPTION
**Pull request by schen32**
- new Friends page to see a list of users
- click on a user to see their habits

### CODE REVIEW

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/FriendsFragment.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        String userUsername;
+                        String userFirstName;
+                        String userLastName;
+                        for (QueryDocumentSnapshot document : task.getResult()) {
+                            userUsername = (String) document.getId();
+                            userFirstName = (String) document.get("FirstName");
+                            userLastName = (String) document.get("LastName");
+                            if (!userUsername.equals(username)) {
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/FriendsFragment.java:106]: Display all users other than self

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java [hunk 2]
```diff
@@ -54,28 +54,27 @@ protected void onCreate(Bundle savedInstanceState) {
             startActivity(intent);
         }
 
-
-        ProfileFragment profileFragment = new ProfileFragment();
-        HabitFragment habitFragment = new HabitFragment();
-
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java:60]: Clicking the back button in an activity gets you back to the fragment you entered the activity from

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java [hunk 3]
```diff
@@ -54,28 +54,27 @@ protected void onCreate(Bundle savedInstanceState) {
...
                     break;
             }
+            getSupportFragmentManager().beginTransaction().replace(R.id.nav_fragment, newFragment).commit();
             return true;
         });
-        if (selected == 0) {
-            bottomNavigationView.setSelectedItemId(R.id.habit);
-            selected = R.id.habit;
-        }
+
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java:74]: Use an intent/extra to specific which fragment in MainActivity you wish to go back to

File: CMPUT301Project/app/src/main/res/menu/bottom_nav_menu.xml [hunk 4]
```diff
@@ -13,4 +13,8 @@
         android:id="@+id/profile"
         android:title="Profile"
         android:icon="@drawable/ic_person_foreground"/>
+    <item
+        android:id="@+id/friends"
+        android:title="Friends"
+        android:icon="@drawable/ic_person_foreground"/>
```
Comment (marty-rudolf) [inline CMPUT301Project/app/src/main/res/menu/bottom_nav_menu.xml:19]: We should have a different icon here. Both profile and friends are ic_person_foreground

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java [hunk 5]
```diff
@@ -54,28 +54,27 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                    newFragment = new HabitFragment();
+                    break;
+                case R.id.friends:
+                    newFragment = new FriendsFragment();
                     break;
-                case R.id.profile:
-                    selected = R.id.profile;
-                    getSupportFragmentManager().beginTransaction().replace(R.id.nav_fragment, profileFragment).commit();
+                default:
+                    newFragment = new ProfileFragment();
```
Comment (marty-rudolf) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java:68]: Possibly better to have this as another case and have default do nothing?

**Review by marty-rudolf — APPROVED** [review]

File: CMPUT301Project/app/src/main/res/menu/bottom_nav_menu.xml (hunk 4 above)
Comment (schen32) [reply CMPUT301Project/app/src/main/res/menu/bottom_nav_menu.xml:19]: I temporarily used this icon and forgot to change it. We can change it in another update

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java (hunk 5 above)
Comment (schen32) [reply CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java:68]: We still want to default to a fragment and Android Studio complains if I have a redundant case for profile

File: CMPUT301Project/app/src/main/res/menu/bottom_nav_menu.xml (hunk 4 above)
Comment (marty-rudolf) [reply CMPUT301Project/app/src/main/res/menu/bottom_nav_menu.xml:19]: I will look into this.

**Pull request merged**